File: functions/openssl/variants/c/3__booth_recode_w5_9.c
#include <stdint.h>

static unsigned int _booth_recode_w5(unsigned int in) 
{
    unsigned int s, d;

    s = ~(((int)(in & 0x20U)) >> 5U);
    s = s + 1U;
    d = 63U - in;
    d = (d & s) | (in & ~s);
    d = (d + (d & 1U)) >> 1U;

    return (d << 1U) | (s & 1U);
}


int main(){
    _booth_recode_w5(0);
}

File: functions/openssl/variants/go/3__booth_recode_w5_9.go
package main

func boothRecodeW5(in uint32) uint32 {
    s := ^((in >> 5) - 1)
    d := (1 << 6) - in - 1
    d |= in & (^s)
    d = (d >> 1) + (d & 1)

    return (d << 1) + (s & 1)
}


func main(){
    boothRecodeW5(0)
}
